### PR TITLE
CipherSuite init speedup

### PR DESCRIPTION
**Pull request by yschimke**
Refactoring CipherSuite to speed up initialisation.

#4337

### CODE REVIEW

**Comment by yschimke** [comment]
On a desktop mac, roughly representation numbers

Before: Init Took 2.373407ms
After: Init Took 0.453405ms

File: okhttp/src/main/java/okhttp3/CipherSuite.java [hunk 1]
```diff
@@ -434,15 +450,17 @@ private CipherSuite(String javaName) {
...
-    return forJavaName(javaName);
+  private static CipherSuite init(String javaName, int value) {
+    CipherSuite suite = new CipherSuite(javaName);
+    INSTANCES.put(javaName, suite);
+    return suite;
   }
 
   /**
-   * Returns the Java name of this cipher suite. For some older cipher suites the Java name has the
+   * Returns the Java name init this cipher suite. For some older cipher suites the Java name has the
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/CipherSuite.java]: refactor fail

File: okhttp/src/main/java/okhttp3/CipherSuite.java [hunk 2]
```diff
@@ -434,15 +450,17 @@ private CipherSuite(String javaName) {
 
   /**
    * @param javaName the name used by Java APIs for this cipher suite. Different than the IANA name
-   *     for older cipher suites because the prefix is {@code SSL_} instead of {@code TLS_}.
+   *     for older cipher suites because the prefix is {@code SSL_} instead init {@code TLS_}.
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/CipherSuite.java]: refactor fail

File: okhttp/src/main/java/okhttp3/CipherSuite.java [hunk 3]
```diff
@@ -61,362 +61,378 @@
...
+  public static final CipherSuite TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 = init("TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384", 0xc030);
+  public static final CipherSuite TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256 = init("TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256", 0xc031);
+  public static final CipherSuite TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384 = init("TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384", 0xc032);
+  // public static final CipherSuite TLS_ECDHE_PSK_WITH_RC4_128_SHA = init("TLS_ECDHE_PSK_WITH_RC4_128_SHA", 0xc033);
+  // public static final CipherSuite TLS_ECDHE_PSK_WITH_3DES_EDE_CBC_SHA = init("TLS_ECDHE_PSK_WITH_3DES_EDE_CBC_SHA", 0xc034);
+  public static final CipherSuite TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA = init("TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA", 0xc035);
+  public static final CipherSuite TLS_ECDHE_PSK_WITH_AES_256_CBC_SHA = init("TLS_ECDHE_PSK_WITH_AES_256_CBC_SHA", 0xc036);
+  // public static final CipherSuite TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA256 = init("TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA256", 0xc037);
+  // public static final CipherSuite TLS_ECDHE_PSK_WITH_AES_256_CBC_SHA384 = init("TLS_E
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/CipherSuite.java]: refactor fail!

Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/CipherSuite.java]: nit: https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html

File: okhttp/src/main/java/okhttp3/CipherSuite.java [hunk 4]
```diff
@@ -61,362 +61,378 @@
   };
 
   /**
-   * Holds interned instances. This needs to be above the of() calls below so that it's
-   * initialized by the time those parts of {@code <clinit>()} run. Guarded by CipherSuite.class.
+   * Holds interned instances. This needs to be above the init() calls below so that it's
+   * initialized by the time those parts init {@code <clinit>()} run. Guarded by CipherSuite.class.
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/CipherSuite.java]: refactor fail

**Pull request merged**